File: java-crm/shared-kernel/src/main/java/com/eduplatform/shared/paging/Order.java
package com.eduplatform.shared.paging;

import lombok.Data;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
public class Order {
    private String property;
    private Direction direction;

    public Order() {
    }

    public Order(String property, Direction direction) {
        this.property = property;
        this.direction = direction;
    }

    public enum Direction {
        ASC, DESC
    }
}

File: go-services/social-service/internal/paging/order.go
package paging

// Direction represents sort direction
type Direction string

const (
	ASC  Direction = "ASC"
	DESC Direction = "DESC"
)

// Order represents a sort order
type Order struct {
	Property  string    `json:"property"`
	Direction Direction `json:"direction"`
}

// NewOrder creates a new Order
func NewOrder(property string, direction Direction) Order {
	return Order{
		Property:  property,
		Direction: direction,
	}
}
